src/app/cards/guards/card-exists.guard.ts

```diff
@@ -0,0 +1,42 @@
+import { Injectable } from '@angular/core';
+import { ActivatedRouteSnapshot, CanActivate } from '@angular/router';
+import { Observable, of, zip } from 'rxjs';
+import { catchError, filter, map, switchMap, take, tap } from 'rxjs/operators';
+
+import { Store } from '@ngrx/store';
+
+import * as fromStore from '../store';
+
+@Injectable()
+export class CardExistsGuard implements CanActivate {
+  constructor(private store: Store<fromStore.CardsState>) {}
+
+  canActivate(route: ActivatedRouteSnapshot): Observable<boolean> {
+    const cardId = route.paramMap.get('cardId');
+    this.store.dispatch(fromStore.setSelectedCardId({ cardId }));
+    return this.checkCards(cardId).pipe(
+      switchMap(() => of(true)),
+      catchError(() => of(false))
+    );
+  }
+
+  private cardExists(cardId: string): Observable<boolean> {
+    return this.store.select(fromStore.selectCardEntities).pipe(
+      map((entities) => !!entities[cardId]),
+      take(2) // Once for checking if the card exists, the second to give the card a chance to load
```

Comment:
The flow goes something like this:
1. dispatch the selected card id
1. check if that card exists
1. if it doesn't exist, dispatch the `loadCards` action
1. if it exists now, return true.